appi.adoc

```diff
@@ -0,0 +1,397 @@
+
+. Each group's attributes, dimensions and variables should be self-contained in combination with the group metadata and any inherited properties (e.g., dimension sizes, coordinates).
+This ensures that if a group is renamed or extracted (with any inherited properties) into a new file, its information content is preserved.
+For example, storage of ensembles as sibling groups (as described below) is often clearer when the realization number is encoded in the group name.
+In this case, the `Realization` group attribute retains the realization number even if the group is renamed.
+Other commonly enumerated group names, such as station identifiers, buoy numbers, or channel wavelengths are analogous.
+It is fine to enumerate or itemize names so long as the number or name is redundantly stored as a group attribute.
+
+. Renaming or moving a group or self-contained branch of groups to a new location should not affect the interpretation of data.
+Relative and absolute paths (containing "/") of coordinates, dimensions, and attributes are fragile in that they may break if groups are moved or renamed.
+To prevent this, it is preferred that paths not be present in attribute values.
+Instead, named objects resolve to the most proximal object (i.e., dimension or variable) of that name that has the referring attribute within its scope.
+This makes files which use groups amenable to sub-setting and aggregation.
+For example, the CF `coordinates` attribute identifies a variable's coordinates in a whitespace-separated list such as `"lat lon"`.
+When the coordinates are outside the group that contains the `coordinates` attribute, it is tempting to store the coordinate locations as full, unambiguous paths such as `"/g1/lat /g1/lon"`, or as relative paths such as `"g1/lat g1/lon"`.
+However, paths that contain slashes must be explicitly altered when the variable is subset into a new file with a different group hierarchy, or when the hierarchy is flattened.
+CF attributes affected by this practice include `ancillary_variables`, `bounds`, `cell_measures`, `climatology`,  `coordinates`, `formula_terms`, and `grid_mapping`.
+The simpler `"lat lon"` specification works without alteration in all situations for out-of-group locations so long as it is understood to mean the nearest identifiers that have the referring variable in their scope.
+
+. When using a grid mapping which is referenced in multiple groups, the reference to the coordinates themselves should be maintained as an attribute of the referencing variables, and not as an attribute of the grid mapping itself. This way the coordinates can resolve to the correct coordinates which are "owned" by the referencing variable, and which are not shared amongst variables, while the grid mapping which is used to interpret the coordinates is stored in only one place. Both coordinates and grid mapping are thus found via regular scoping mechanisms.
++
```

Comment:
I don't really get this, I'm afraid. CF doesn't let you store coordinate variables on a grid mapping variable, anyway, so what is the need for this advice, here?

Reply:
For me the advantage here is extra clarity. There are some cases where the same grid mapping is re-used, but the coordinates are different, for example if you're re-using an orthographic coordinate system in a netCDF file with radiances from different instruments viewing with different resolutions but on the same platform - basically two variables might share `grid_mapping_name` but have different variables storing `projection_x_coordinate` and `projection_x_coordinate`.

Our teams found this clearer but perhaps this is just over-specifying. If this principal is clear without this clause I'm happy to drop the clause.

Reply:
OK, I see where you're coming from. I think you're right that it is worth highlighting that groups don't change the standard behaviour here. Perhaps this minor change would make it clear that there is nothing surprising here? 

"When using a grid mapping which is referenced in multiple groups, the reference to the
coordinates themselves should be maintained as an attribute of the referencing variables,
**in the usual manner (ref to section 5.6)**. This way the coordinates can resolve to the correct
coordinates which are "owned" by the referencing variable, and which are not shared amongst
variables, while the grid mapping which is used to interpret the coordinates is stored in only one place. Both coordinates and grid mapping are thus found via regular scoping mechanisms."